keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/resource/TitusBaseClusterHandlerTests.kt

```diff
@@ -0,0 +1,117 @@
+package com.netflix.spinnaker.keel.titus.resource
+
+import com.netflix.spinnaker.keel.api.Moniker
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceDiff
+import com.netflix.spinnaker.keel.api.SimpleLocations
+import com.netflix.spinnaker.keel.api.SimpleRegionSpec
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.ec2.Capacity
+import com.netflix.spinnaker.keel.api.plugins.BaseClusterHandlerTests
+import com.netflix.spinnaker.keel.api.plugins.Resolver
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.keel.api.titus.TitusClusterSpec
+import com.netflix.spinnaker.keel.api.titus.TitusServerGroup
+import com.netflix.spinnaker.keel.api.titus.TitusServerGroupSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.diff.DefaultResourceDiff
+import com.netflix.spinnaker.keel.docker.DigestProvider
+import com.netflix.spinnaker.keel.orca.ClusterExportHelper
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.titus.TITUS_CLUSTER_V1
+import com.netflix.spinnaker.keel.titus.TitusClusterHandler
+import com.netflix.spinnaker.keel.titus.byRegion
+import com.netflix.spinnaker.keel.titus.resolve
+import io.mockk.mockk
+import io.mockk.spyk
+import java.time.Clock
+
+class TitusBaseClusterHandlerTests : BaseClusterHandlerTests<TitusClusterSpec, TitusServerGroup, TitusClusterHandler>() {
```

Comment:
Why does this need to be a new test class? Can't we reuse `TitusClusterHandlerTests`?

Reply:
`TitusClusterHandlerTests` is super duper complicated, and I don't want to add to it. I want to test this specific functionality in its own test because it is the implementation of the base tests. Same comment with both classes

Reply:
I hear you. I think we have a tendency to structure test classes to match the scope of an entire target class that's being tested, but there's no rule around that. I'm not opposed to breaking up large/complex test classes into smaller ones that are easier to reason about. The only downside I can think of is that distributed is always harder to find than centralized, so people might get confused about what test class to add to, so we should be mindful about the names of the test classes to make that as obvious as possible. Carry on. 🙂

Reply:
I'm all for breaking tests up around behavior. I'd like to also break the handlers up as I think they do not do "just one thing" but bloated tests cause more direct pain.